editor/modes/visual-editor/block.js

```diff
@@ -210,7 +251,13 @@ class VisualEditorBlock extends Component {
 
 		// Deselect on escape.
 		if ( ESCAPE === keyCode ) {
-			onDeselect();
+			if ( isToolbar( target ) ) {
```

Comment:
A few alternative ideas to reaching up through parent ancestry from the event target:

1) If what we're looking for already exists in the rendered element, we could check whether the target is contained within one of them. Something like:

```js
const isToolbar = [
	this.nodes.toolbar,
	this.nodes.blockSettings,
	this.nodes.blockMover
].some( ( node ) => node.contains( event.target ) )
```

2) Maybe it should be the toolbar, block settings, and block mover which handle the keypress and surface this up to `VisualEditorBlock` through a prop callback like `onEscape`.

3) `onFocus` could be called anywhere, including the toolbar, block settings, and block mover components, on their own "on escape press" handling

My preference is toward the second: For behaviors which are specific to component(s), it's nice to implement those behaviors within the component itself, and surface them to the parent component semantically through named prop callbacks.